packages/monorepo-tools/src/sort-workspaces.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-console */
+
+import { promises as fs } from 'fs';
+import path from 'path';
+import { getPackagesInTopologicalOrder } from './utils/get-packages-in-topological-order';
+import { findMonorepoRoot } from './utils/find-monorepo-root';
+
+process.on('unhandledRejection', (err: Error) => {
+  console.error();
+  console.error(err?.stack || err?.message || err);
+  process.exitCode = 1;
+});
+
+async function main() {
+  const monorepoRoot = await findMonorepoRoot();
+  const text = await fs.readFile(
+    path.join(monorepoRoot, 'package.json'),
+    'utf8'
+  );
+  const packageJSON = JSON.parse(text);
+  const patterns: string[] | undefined = packageJSON.workspacesPatterns
+
+  if (!patterns) {
+    console.log('Refusing to sort workspaces without workspacesPatterns in monorepo root package.json');
+    return;
+  }
+
+  const packages = await getPackagesInTopologicalOrder(monorepoRoot, patterns);
+
+  packageJSON.workspaces = packages.map(({ location }) =>
+    location.replace(`${monorepoRoot}/`, '')
```

Comment:
Actually thinking about this now, this will probably not work on windows 🤔

Reply:
Maybe https://nodejs.org/api/path.html#pathrelativefrom-to does the tr
ick here?

Reply:
Turns out if I copy all the fixes it is already the path component returned by glob which is exactly what we need.